soc/espressif/esp32s3/default.ld

```diff
@@ -771,13 +772,9 @@ SECTIONS
 
     _text_end = ABSOLUTE(.);
     _instruction_reserved_end = ABSOLUTE(.);  /* This is a symbol marking the flash.text end, this can be used for mmu driver to maintain virtual address */
+    __text_region_end = ABSOLUTE(.);
     _etext = .;
 
-    /* Similar to _iram_start, this symbol goes here so it is
-     * resolved by addr2line in preference to the first symbol in
-     * the flash.text segment.
-     */
-    //_flash_cache_start = ABSOLUTE(0);
```

Comment:
seems to do a little more than adding `__text_region_start` & `__text_region_end`? I'd prefer splitting it into two commits instead

Reply:
Those deleted lines were already removed from the other SoCs ld script, I forgot to polish the original PR so did it here. But if you insist, I can split :)

Reply:
Code removal might not be as bad, but other line modifications / addition can confuse the next person without the context that we have here and will probably wonder what this has to do with add missing linker symbols

Reply:
The other changes are related to the same topic of providing the valid symbols, therefore they are in this PR.

Reply:
Fair enough